src/reducetensor.cpp

```diff
@@ -1031,6 +1031,10 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
                                        std::to_string(static_cast<int>(use_padding.first)) +
                                        std::to_string(static_cast<int>(use_padding.second));
 
+        int alignment = sizeof(int);
+        Data_t workspace_aligned = reinterpret_cast<Data_t>(
+            (reinterpret_cast<unsigned long>(workspace) + alignment - 1) / alignment * alignment);
+
```

Comment:
Probably `std::align` is better?
Btw, `reinterpret_cast<unsigned long>(workspace)` does truncation on windows, since `sizeof(unsigned long) == 4`
@apwojcik

Also I have doubts that `sizeof(int)` is the best alignment. Probably it works, but 256byte alignment could be better from a performance point of view. May I ask you to double check this?

Reply:
The main reason is to ensure the correctness of the kernel.

I've also recommended using std::align.

Reply:
For reference, Windows data ranges: https://learn.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=msvc-170 (applies to HIP compiler, too).
Like @CAHEK7 mentioned, `unsigned long` is 32bit on Windows (even on a 64bit operating system) but 64bit on Linux. Please use `unsigned long long`, which size is 64bit on both Windows and Linux.

Reply:
Or `uintptr_t` (it is optional, but widely supported). Or `std::align`.

Reply:
Updated, please check,  Thanks